internal/server/iigointernal/legislature.go

```diff
@@ -0,0 +1,166 @@
+package iigointernal
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/voting"
+	"github.com/pkg/errors"
+)
+
+type legislature struct {
+	SpeakerID     shared.ClientID
+	budget        shared.Resources
+	judgeSalary   shared.Resources
+	ruleToVote    string
+	ballotBox     voting.BallotBox
+	votingResult  bool
+	clientSpeaker roles.Speaker
+}
+
+// returnJudgeSalary returns the salary to the common pool.
```

Comment:
ditto above--where commonpool?

Reply:
This returns the salary to the common pool in the case that there are not enough common pool resources for IIGO to run but the salary has already been withdrawn.

Reply:
I was confused when I saw that it returns to the commonpool but commpool was not referenced inside the function